tests/behavioural/features/steps/common.py

```diff
@@ -54,5 +54,12 @@ def step_impl_reuse_the_nth_sent_message(context, message_index):
 @then("'{message_count}' messages are returned")
 def step_impl_n_messages_returned(context, message_count):
     """ validate that the correct number of messages was returned"""
-    nose.tools.assert_equal(int(message_count), len(
-        context.bdd_helper.messages_responses_data[0]['messages']))
+    nose.tools.assert_equal(int(message_count), len(context.bdd_helper.messages_responses_data[0]['messages']))
+
+
+@given("debug step")
+@when("debug step")
+@then("debug step")
+def step_impl_debug_step(context):
```

Comment:
Does this still need to be here?

Reply:
Can be dropped , but very useful when debugging features . I have written it and taken it out about 5 times before this . Thought I would leave it in in case others find it useful

Reply:
I think if we paid for pycharm it may allow debugging of features , but since we don't we can add a debug step in a feature file and breakpoint there . This is needed because the bdd tests reuse steps in many features , so break pointing a step is a pain .

Reply:
Works for me, a comment explaining that would be useful but not essential